### PR TITLE
[GPII-3718] Add alert about missing CouchDB cluster member

### PR DESCRIPTION
**Pull request by amatas**
This PR depends on https://github.com/gpii-ops/couchdb-statefulset-assembler/pull/4 to be effective.

We can get alerts with the following information:
image
which pod can not see which CouchDB node.

This PR also increases the duration of the CouchDB error messages that appear every time that a new ephemeral cluster is created or a node upgrade is performed.

### CODE REVIEW

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_membership_errors.tf [hunk 1]
```diff
@@ -7,7 +7,7 @@ resource "google_monitoring_alert_policy" "couchdb_membership_errors" {
       filter          = "metric.type=\"logging.googleapis.com/user/couchdb_membership.error\" resource.type=\"k8s_container\""
       comparison      = "COMPARISON_GT"
       threshold_value = 1.0
-      duration        = "0s"
+      duration        = "3600s"
```
Comment (natarajaya) [inline gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_membership_errors.tf]: 1 hour duration?

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      duration        = "600s"
+
+      aggregations {
+        alignment_period   = "60s"
+        per_series_aligner = "ALIGN_SUM"
+        group_by_fields    = []
+      }
+
+      denominator_filter       = ""
+      denominator_aggregations = []
```
Comment (stepanstipl) [inline gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf]: Any reason for these to be here?

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    condition_threshold {
+      filter          = "metric.type=\"logging.googleapis.com/user/couchdb_missing_node.error\" resource.type=\"k8s_container\""
+      comparison      = "COMPARISON_GT"
+      threshold_value = 0
+      duration        = "600s"
+
+      aggregations {
+        alignment_period   = "60s"
+        per_series_aligner = "ALIGN_SUM"
+        group_by_fields    = []
```
Comment (stepanstipl) [inline gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf]: I think this is not needed, and won't have any effect unless `cross_series_reducer` is udes.

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    display_name = "CouchDB node is missing in the cluster"
+  }
+
+  documentation = {
+    content   = "[Use this link to explore policy events in Logs Viewer](https://console.cloud.google.com/logs/viewer?project=${var.project_id}&organizationId=247149361674&supportedpurview=project&minLogLevel=0&expandAll=false&limitCustomFacetWidth=true&advancedFilter=resource.type%3D%22k8s_container%22%20AND%20resource.labels.project_id%3D%22${var.project_id}%22%20AND%20resource.labels.cluster_name%3D%22k8s-cluster%22%20AND%20resource.labels.namespace_name%3D%22gpii%22%20AND%20labels.k8s-pod%2Fapp%3D%22couchdb%22%20AND%20labels.k8s-pod%2Frelease%3D%22couchdb%22%20AND%20resource.labels.container_name%3D%22couchdb-statefulset-assembler%22%20AND%20textPayload:%20%22%20couchdb%20node%20in%20the%20cluster:%22)"
+    mime_type = "text/markdown"
+  }
+
+  notification_channels = ["${google_monitoring_notification_channel.email.name}", "${google_monitoring_notification_channel.alerts_slack.*.name}"]
+  user_labels           = {}
```
Comment (stepanstipl) [inline gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf]: Any reason for these to be here if it's empty?

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_membership_errors.tf (hunk 1 above)
Comment (amatas) [reply gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_membership_errors.tf]: image

Yes, 30 minutes are not enough for a node upgrade, so the next step is 1 hour.

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf (hunk 4 above)
Comment (amatas) [reply gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf]: came from C&P from other alerts, I deleted all of them in all alerts.

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf (hunk 3 above)
Comment (amatas) [reply gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf]: came from C&P from other alerts, I deleted all of them in all alerts.

File: gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf (hunk 2 above)
Comment (amatas) [reply gcp/modules/gcp-stackdriver-monitoring/alert_couchdb_cluster_missing_node.tf]: came from C&P from other alerts, I deleted all of them in all alerts.

**Comment by stepanstipl** [comment]
Thanks @amatas! 

I have the same question as in #566 - do we have any empirical evidence that the newly introduced wait logic, that was previously removed from #566 as it "seems that does not make difference with or without it", does actually improve things?

And how does it differ to what Terraform provider already seems to be doing (reading metric after creating it)?

Otherwise good job with the improved alerting logic 👍

**Comment by amatas** [comment]
> Thanks @amatas!
> 
> I have the same question as in #566 - do we have any empirical evidence that the newly introduced wait logic, that was previously removed from #566 as it "seems that does not make difference with or without it", does actually improve things?
> 
> And how does it differ to what Terraform provider already seems to be doing (reading metric after creating it)?
> 
> Otherwise good job with the improved alerting logic +1
```
google_logging_metric.disks_createsnapshot: Creating...
  filter:                                  "" => "resource.type=\"gce_disk\" AND protoPayload.methodName=\"v1.compute.disks.createSnapshot\""
  label_extractors.%:                      "" => "2"
  label_extractors.pv_name:                "" => "REGEXP_EXTRACT(protoPayload.request.description, \".*\\\"name.:..([^\\\"]+).*\")"
  label_extractors.severity:               "" => "EXTRACT(severity)"
  metric_descriptor.#:                     "" => "1"
  metric_descriptor.0.labels.#:            "" => "2"
  metric_descriptor.0.labels.0.key:        "" => "pv_name"
  metric_descriptor.0.labels.0.value_type: "" => "STRING"
  metric_descriptor.0.labels.1.key:        "" => "severity"
  metric_descriptor.0.labels.1.value_type: "" => "STRING"
  metric_descriptor.0.metric_kind:         "" => "DELTA"
  metric_descriptor.0.unit:                "" => "1"
  metric_descriptor.0.value_type:          "" => "INT64"
  name:                                    "" => "compute.disks.createSnapshot"
  project:                                 "" => "gpii2test-gcp-dev-doe"
google_logging_metric.disks_createsnapshot: Creation complete after 1s (ID: compute.disks.createSnapshot)
google_monitoring_alert_policy.disk_snapshots: Creating...
[...]
Error: Error applying plan:
1 error occurred:
	* google_monitoring_alert_policy.disk_snapshots: 1 error occurred:
	* google_monitoring_alert_policy.disk_snapshots: Error creating AlertPolicy: googleapi: Error 404: The metric referenced by the provided filter is unknown. Check the metric name and labels.
```

Someone is lying here either Terraform or Google API. The only simple way that comes to my mind about how to detect if a lbm is created is using the gcloud command.

**Pull request merged**